### PR TITLE
Bigtable: non-functional improvements relating to error handling

### PR DESCRIPTION
**Pull request by igorbernstein2**
* Add error handling examples to the javadocs
* Use gax for error handling now that the code has been upstreamed
* Fix outdated bigtable-admin readme

### CODE REVIEW

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java [hunk 1]
```diff
@@ -160,7 +159,7 @@ public void close() {
    */
   @SuppressWarnings("WeakerAccess")
   public Instance createInstance(CreateInstanceRequest request) {
-    return awaitFuture(createInstanceAsync(request));
+    return ApiExceptions.callAndTranslateApiException(createInstanceAsync(request));
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: Hope this change won't cause some test to fail.

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 2]
```diff
@@ -159,6 +180,19 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
    *   String tableId = "[TABLE]";
    *
    *   ApiFuture<Row> result = bigtableClient.readRow(tableId,  ByteString.copyFromUtf8("key"));
+   *
+   *   ApiFutures.addCallback(futureResult, new ApiFutureCallback<Row>() {
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: result or futureResult?

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 3]
```diff
@@ -417,9 +547,23 @@ public BulkMutationBatcher newBulkMutationBatcher() {
...
    *   BulkMutation batch = BulkMutation.create("[TABLE]");
    *   for (String someValue : someCollection) {
-   *     batch.add("[ROW KEY]", Mutation.create().setCell("[FAMILY NAME]", "[QUALIFIER]", "[VALUE]");
+   *     ApiFuture<Void> entryFuture = batch.add("[ROW KEY]",
+   *       Mutation.create().setCell("[FAMILY NAME]", "[QUALIFIER]", "[VALUE]"));
    *   }
-   *   ApiFuture<Void> result = bigtableClient.bulkMutateRowsAsync(batch);
+   *   ApiFuture<Void> resultFuture = bigtableClient.bulkMutateRowsAsync(batch);
+   *
+   *   ApiFutures.addCallback(future, new ApiFutureCallback<Void>() {
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: resultFuture or future?

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 4]
```diff
@@ -487,7 +645,14 @@ public BulkMutationBatcher newBulkMutationBatcher() {
    *         .setCell("[FAMILY]", "[QUALIFIER]", "[VALUE]")
    *       );
    *
-   *   boolean success = bigtableClient.checkAndMutateRowCallable().call(mutation);
+   *   // Sync style
+   *   try {
+   *     boolean success = bigtableClient.checkAndMutateRowCallable().call(mutation);
+   *   } catch (NotFoundException e) {
+   *     System.out.println("Tried to mutate a non-existent table");
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: row doesn't exist or table doesn't?

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 5]
```diff
@@ -465,6 +610,19 @@ public BulkMutationBatcher newBulkMutationBatcher() {
    *       );
    *
    *   ApiFuture<Boolean> future = bigtableClient.checkAndMutateRowAsync(mutation);
+   *
+   *   ApiFutures.addCallback(future, new ApiFutureCallback<Boolean>() {
+   *     public void onFailure(Throwable t) {
+   *       if (t instanceof NotFoundException) {
+   *         System.out.println("Tried to mutate a non-existent table");
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: row or table doesn't exist?

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java (hunk 1 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: It shouldn't. The implementation in callAndTranslateApiException is based on awaitFuture

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 2 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: fixed

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 3 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: fixed

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 4 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: table, I extended the example to show how unmatched rows behave

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 5 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: table

Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: table

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 4 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: table

**Comment by igorbernstein2** [comment]
Thanks for the quick review. All feedback should be addressed

**Comment by igorbernstein2** [comment]
@pongad This should be ready for merging

**Pull request merged**